docs/case_search_query_language.rst

```diff
@@ -0,0 +1,449 @@
+
+
+Filtering on parent (ancestor) cases
+====================================
+
+Searches may be performed against ancestor cases (e.g. parent cases) using the ``/`` operator
+
+.. code-block::
+
+    # search for cases that have a 'parent' case that matches the filter 'age > 55'
+    parent/age > 55
+
+    # successive steps can be added to navigate further up the case hierarchy
+    parent/parent/dod = ''
+
+``ancestor-exists``
+-------------------
+* **Behavior**: Match cases that have an ancestor with the given relation that matches the ancestor
+  filter expression.
+* **Arguments**: Two arguments, the ancestor relationship (usually one of parent or host) and the
+  ancestor filter expression.
+* **Usage**:
+    * ``ancestor-exists(parent/parent, city = 'SF')``
+    * ``ancestor-exists(parent, food_included = 'yes' and ancestor-exists(parent, city!='' and
+      selected(city, 'Boston')))``
+* **Limitation**:
+    * The arguments can't be a standalone function and must be a binary expression
+        * This will *not* work: ``ancestor-exists(parent, selected(city, 'SF'))``
+        * This will work:  ``ancestor-exists(parent, city != '' and selected(city, 'SF'))``
+    * The ancestor filter expression may not include ``subcase-exists`` or ``subcase-count``
+      * **Best Practices**:
+    * Limit multiple uses of this function in your query to avoid performance implications
+    * Add as many arguments to ``ancestor-exists()`` as possible to help narrow down results (i.e.,
+      by ``@case_type`` or ``@status``)
+
+
+Filtering on child cases (subcases)
+===================================
+
+Special functions are provided to support filtering based on the properties of subcases. These are:
+
+``subcase-exists``
+------------------
+* **Behavior**: Match cases that have a subcase with the given relation that matches the subcase
+  filter expression.
+* **Arguments**: Two arguments, the subcase relationship (usually one of 'parent' or 'host') and the
+  subcase filter expression.
+* **Usage**: ``subcase-exists('parent', lab_type = 'blood' and result = 1)``
+* **Best Practices**:
+    * Limit multiple uses of this function in your query to avoid performance implications
+    * Add as many arguments to ``subcase-exists()`` as possible to help narrow down results (i.e.,
+      by ``@case_type`` or ``@status``)
+
+``subcase-count``
+-----------------
+* **Behavior**: Match cases where the number of subcases matches the given expression.
+* **Arguments**: Two arguments, the subcase relationship (usually one of 'parent' or 'host') and the
+  subcase filter expression.
+* **Usage**: ``subcase-count('parent', lab_type = 'blood' and result = 1) > 3``
+    * The count function must be used in conjunction with a comparison operator. All operators are
+      supported (``=``, ``!=``, ``<``, ``<=``, ``>``, ``>=``)
+* **Best Practices**:
+    * Limit multiple uses of this function in your query to avoid performance implications
+
+.. warning::
```

Comment:
This section applies to all subcase and ancestor queries. What about moving it to the top of a "Filtering on related cases" section?